pdata/internal/json/number.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json // import "go.opentelemetry.io/collector/pdata/internal/json"
+
+import (
+	"strconv"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+// ReadInt32 unmarshalls JSON data into an int32. Accepts both numbers and strings decimal.
+// See https://developers.google.com/protocol-buffers/docs/proto3#json.
+func ReadInt32(iter *jsoniter.Iterator) int32 {
+	switch iter.WhatIsNext() {
+	case jsoniter.NumberValue:
+		return iter.ReadInt32()
+	case jsoniter.StringValue:
+		val, err := strconv.ParseUint(iter.ReadString(), 10, 32)
```

Comment:
To handle negative numbers correctly:
```suggestion
		val, err := strconv.ParseInt(iter.ReadString(), 10, 32)
```
Also would be good to add a test for negative numbers.

Reply:
Will do.